language-plutus-core/src/Language/PlutusCore/Constant/Apply.hs

```diff
@@ -122,118 +107,77 @@ instance ( PrettyBy config (Constant ())
-    -> SizeValues
+    => TypeScheme a r
     -> Value TyName Name ()
-    -> Evaluate m (ConstAppResult (a, SizeValues))
-extractSchemed (TypeSchemeBuiltin a)   sizeValues value = extractBuiltin a sizeValues value
-extractSchemed (TypeSchemeArrow _ _)   _          _     = error "Not implemented."
-extractSchemed (TypeSchemeAllType _ _) _          _     = error "Not implemented."
-extractSchemed (TypeSchemeAllSize _)   _          _     = error "Not implemented."
+    -> Evaluate m (ConstAppResult a)
+extractSchemed (TypeSchemeBuiltin a)   value = extractBuiltin a value
+extractSchemed (TypeSchemeArrow _ _)   _     = error "Not implemented."
+extractSchemed (TypeSchemeAllType _ _) _     = error "Not implemented."
 
 -- | Apply a function with a known 'TypeScheme' to a list of 'Constant's (unwrapped from 'Value's).
 -- Checks that the constants are of expected types and there are no size mismatches.
 applyTypeSchemed
     :: Monad m
-    => TypeScheme SizeVar a r -> a -> [Value TyName Name ()] -> Evaluate m ConstAppResultDef
-applyTypeSchemed schema = go schema (SizeVar 0) (SizeValues mempty) where
+    => TypeScheme a r -> a -> [Value TyName Name ()] -> Evaluate m ConstAppResultDef
+applyTypeSchemed = go where
     go
         :: Monad m
-        => TypeScheme SizeVar a r
-        -> SizeVar
-        -> SizeValues
+        => TypeScheme a r
         -> a
         -> [Value TyName Name ()]
         -> Evaluate m ConstAppResultDef
-    go (TypeSchemeBuiltin tb)      _       sizeValues y args = case args of  -- Computed the result.
+    go (TypeSchemeBuiltin tb)      y args = case args of  -- Computed the result.
         -- This is where all the size checks prescribed by the specification happen.
         -- We instantiate the size variable of a final 'TypedBuiltin' to its value and call
         -- 'makeConstAppResult' which performs the final size check before converting
         -- a Haskell value to the corresponding PLC one.
-        [] -> return . makeConstAppResult $ TypedBuiltinValue (substSizeVar sizeValues tb) y
+        [] -> return . makeConstAppResult $ TypedBuiltinValue tb y
         _  -> return . ConstAppError $ ExcessArgumentsConstAppError args     -- Too many arguments.
-    go (TypeSchemeAllType _ schK)  sizeVar sizeValues f args =
-        go (schK TypedBuiltinDyn) sizeVar sizeValues f args
-    go (TypeSchemeArrow schA schB) sizeVar sizeValues f args = case args of
+    go (TypeSchemeAllType _ schK)  f args =
+        go (schK TypedBuiltinDyn) f args
+    go (TypeSchemeArrow schA schB) f args = case args of
         []          -> return ConstAppStuck             -- Not enough arguments to compute.
         arg : args' -> do                               -- Peel off one argument.
             -- Coerce the argument to a Haskell value.
-            res <- extractSchemed schA sizeValues arg
-            forBind res $ \(x, sizeValues') ->
+            res <- extractSchemed schA arg
+            forBind res $ \x ->
```

Comment:
Dang! This conflicts a lot with my existing PR.

Reply:
Hm, maybe we should have merged yours first. Oh well, I think it's mostly mechanical.